File: dsa-problems/Stacks/solutions/cpp/STK-015-bike-repair-plates.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int countUnsafe(vector<int>& d) {
        int count = 0;
        int n = d.size();
        for (int i = 0; i < n - 1; i++) {
            if (d[i+1] > d[i]) {
                count++;
            }
        }
        return count;
    }
};

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    
    int n;
    if (!(cin >> n)) return 0;
    
    vector<int> d(n);
    for (int i = 0; i < n; i++) {
        cin >> d[i];
    }
    
    Solution sol;
    cout << sol.countUnsafe(d) << endl;
    
    return 0;
}

File: dsa-problems/Stacks/solutions/java/STK-015-bike-repair-plates.java
import java.util.*;
import java.io.*;

class Solution {
    public int countUnsafe(int[] d) {
        int count = 0;
        for (int i = 0; i < d.length - 1; i++) {
            if (d[i+1] > d[i]) {
                count++;
            }
        }
        return count;
    }
}

class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String line = "";
        while ((line = br.readLine()) != null && line.trim().isEmpty()) {}
        if (line == null) return;
        
        int n = Integer.parseInt(line.trim());
        
        // Read Array (robust)
        List<Integer> list = new ArrayList<>();
        StringTokenizer st = new StringTokenizer(br.readLine());
        
        while (list.size() < n) {
            while (!st.hasMoreTokens()) {
                String l = br.readLine();
                if (l == null) break;
                st = new StringTokenizer(l);
            }
            if (!st.hasMoreTokens()) break;
            list.add(Integer.parseInt(st.nextToken()));
        }
        
        int[] d = new int[list.size()];
        for(int i=0; i<list.size(); i++) d[i] = list.get(i);
        
        Solution sol = new Solution();
        System.out.println(sol.countUnsafe(d));
    }
}
